File: Swift-C++-IR-Comparison/test/controlFlow/Swift/SwitchFallthrough.swift
@main
struct MyMain {

    static func main() {
        let res = switchMe(input: (2, 3))
    }

    static func switchMe(input: (Int, Int)) -> Int {
        var counter = 0
        var max = 7
        switch input {
        case (let x, 0):
            counter = 100
            fallthrough
        case (0, let y):
            counter = -y
        case let (x, y):
            counter = -x
        default:
            counter = 42
        }
        return counter
    }
}

File: Swift-C++-Testsuite/AutoTranspiledTests/try-1/SwitchFallthrough.cpp


struct MyMain {

    static int main() {
        int res = switchMe(2, 3);
        return 0;
    }

    static int switchMe(int x, int y) {
        int counter = 0;
        int max = 7;
        switch(x)
        {
            case(x):
                counter = 100;
            // fallthrough
            case 0:
                counter = -y;
            case(x):
                counter = -x;
            default:
                counter = 42;
        }
        return counter;
    }
};

int main(){
    MyMain::main();
    return 0;
}